storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -201,10 +193,11 @@ impl StateMerklePruner {
         batch_size: usize,
     ) -> Result<(Vec<StaleNodeIndex>, bool)> {
         let mut indices = Vec::new();
-        let mut iter = self
-            .state_merkle_db
-            .iter::<StaleNodeIndexSchema>(ReadOptions::default())?;
-        iter.seek(&start_version)?;
+        let mut iter = self.state_merkle_db.iter::<S>(ReadOptions::default())?;
+        iter.seek(&StaleNodeIndex {
+            stale_since_version: start_version,
+            node_key: NodeKey::new_empty_path(0),
```

Comment:
Why do we change this?

Reply:
good catch.

Reply:
otherwise I need to add `where Version: SeekKey<S>` everywhere